### PR TITLE
Move TestOAuthServerHeaders to e2e

### PR DESCRIPTION
**Pull request by sanchezl**
Part of Issue https://github.com/openshift/origin/issues/23258

### CODE REVIEW

File: test/extended/oauth/server_headers.go [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+	oc = util.NewCLI("oauth-server-headers", util.KubeConfigPath())
+
+	g.BeforeEach(func() {
+		var err error
+
+		transport, err = rest.TransportFor(rest.AnonymousClientConfig(oc.UserConfig()))
+		o.Expect(err).ToNot(o.HaveOccurred())
+
+		// deploy oauth server
+		osinProvider, err := oauthserver.GetRawExtensionForOsinProvider(&osinv1.AllowAllPasswordIdentityProvider{})
```
Comment (stlaz) [inline test/extended/oauth/server_headers.go]: While it does not matter that much for this exact test, the `AllowAll` IdP is not really supported in 4.x anymore, do you think you could set up a simple HTPasswd IdP instead?

File: test/extended/oauth/server_headers.go [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+			checkUrl.Path = path
+			fmt.Fprintf(g.GinkgoWriter, "CheckUrl: %v\n", checkUrl)
+			req, err := http.NewRequest("GET", checkUrl.String(), nil)
+			o.Expect(err).ToNot(o.HaveOccurred())
+
+			req.Header.Set("Accept", "text/html; charset=utf-8")
+
+			// sometimes it takes a few seconds for the oauth server to be ready
+			var resp *http.Response
+			o.Eventually(func() error {
```
Comment (enj) [inline test/extended/oauth/server_headers.go]: Make the OAuth server always be ready.

File: test/extended/oauth/server_headers.go [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+			UseAsLogin:      true,
+		}
+		// deploy oauth server
+		tokenOptions, oauthServerCleanup, err := oauthserver.DeployOAuthServer(oc, []osinv1.IdentityProvider{identityProvider}, nil, []corev1.Secret{secret})
+		defer oauthServerCleanup()
+		o.Expect(err).ToNot(o.HaveOccurred())
+		oauthServerAddr = tokenOptions.Issuer
+	})
+
+	t.DescribeTable("expected headers returned from the",
```
Comment (enj) [inline test/extended/oauth/server_headers.go]: I hope this is a table test.  I have no idea what ginko does...

**Pull request merged**